File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/internal/services/helpers/ListenerRegistration.java
package kin.sdk.internal.services.helpers;

import kin.sdk.internal.services.BlockchainEventsImpl;

/**
 * Represents a listener to {@link BlockchainEventsImpl}, that can be removed using {@link #remove()}.
 */
public class ListenerRegistration {

    private final Runnable removeRunnable;

    public ListenerRegistration(Runnable removeRunnable) {
        this.removeRunnable = removeRunnable;
    }

    /**
     * Remove and unregisters this listener.
     */
    public void remove() {
        removeRunnable.run();
    }
}

File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/internal/services/helpers/ListenerRegistration.kt
package kin.sdk.internal.services.helpers

import kin.sdk.internal.services.BlockchainEventsImpl

/**
 * Represents a listener to [BlockchainEventsImpl], that can be removed using [.remove].
 */
class ListenerRegistration(private val removeRunnable: Runnable) {

    /**
     * Remove and unregisters this listener.
     */
    fun remove() {
        removeRunnable.run()
    }
}
